internal/querynode/impl.go

```diff
@@ -520,7 +517,23 @@ func (node *QueryNode) GetSegmentInfo(ctx context.Context, in *queryPb.GetSegmen
 		return res, nil
 	}
 	infos = append(infos, streamingSegmentInfos...)
-	// log.Debug("GetSegmentInfo: get segment info from query node", zap.Int64("nodeID", node.session.ServerID), zap.Any("segment infos", infos))
+
```

Comment:
suggest defining a filter func with blew code, then use it on streamingSegmentInfos and historicalSegmentInfos slice directly.